webapp/lib/app/configurator/tags/controller_view_helper.dart

```diff
@@ -15,20 +15,22 @@ List<MenuItem> getMenuItems(model.Tag tag) {
   ];
 }
 
-void _addTagsToView(Map<String, List<model.Tag>> tagsByCategory, {bool startEditing = false, bool startEditingName = false}) {
+void _addTagsToView(Map<String, List<model.Tag>> tagsByCategory, Set<String> unsavedTagIds, Set<String> unsavedGroupIds, {bool startEditing = false, bool startEditingName = false}) {
```

Comment:
as discussed, let's move marking the unsaved tags in a separate function.

Processing unsaved tags & groups here complicates these methods even though it is more efficient performance wise, and it's also conceptually 2 operations - one for adding new tags, the other for updating unsaved status.